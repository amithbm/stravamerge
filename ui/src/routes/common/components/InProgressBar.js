import React from 'react'

const InProgressBar = ({inProgress}) => {
    return <div className='inprogresscontainer'>
        <div style={{position: 'absolute', height: '0.3em', width: '25%', backgroundColor: 'rgb(218,235,255)' }} className={inProgress ? 'inprogressanimate inprogressbase' : 'inprogressbase'}></div>
    </div>
}

InProgressBar.propTypes = {
    inProgress: React.PropTypes.bool.isRequired
}

export default InProgressBar