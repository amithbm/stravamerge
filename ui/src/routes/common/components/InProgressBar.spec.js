import React from 'react'
import {shallow} from 'enzyme'
import InProgressBar from './InProgressBar'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<InProgressBar />', () => {
        it('renders as expected', () => {
            const component =  shallow(<InProgressBar inProgress={true} />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})

