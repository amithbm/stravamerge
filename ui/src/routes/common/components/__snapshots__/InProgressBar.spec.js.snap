exports[`components <InProgressBar /> renders as expected 1`] = `
<div
  className="inprogresscontainer">
  <div
    className="inprogressanimate inprogressbase"
    style={
      Object {
        "backgroundColor": "rgb(218,235,255)",
        "height": "0.3em",
        "position": "absolute",
        "width": "25%",
      }
    } />
</div>
`;
